bounded_async_foreach.hpp

```diff
@@ -3,78 +3,78 @@
 #include <functional>
 #include <iostream>
 #include <memory>
+#include <system_error>
 
 // Apply callback to each element in container but limit number of in items
 // being processed.
 // Args:
 //      Container -- any contianer that has cbegin(), cend()
 //      Callback  -- callable object that accepts (const Value& v, Callback
 //      done) ).
-template <typename Container, typename Callback, typename FinishedCallback>
-auto bounded_async_foreach(unsigned n, Container c, Callback cb,
-                           FinishedCallback finished) {
-  using iter_t = decltype(std::cbegin(c));
+template <typename Container, typename Callback, typename FinishCallback>
+void bounded_async_foreach(unsigned n, Container c, Callback cb,
+                           FinishCallback finished_cb) {
+  if (n == 0) {
+    finished_cb(make_error_code(std::errc::no_child_process));
+    return;
+  } else if (std::distance(std::cbegin(c), std::cend(c)) == 0) {
```

Comment:
is it a fancy way to say
`std::empty(c)`

Reply:
is it supposed to be available for all collections?

Reply:
done.